scripts/upgrade

```diff
@@ -24,37 +24,31 @@ upgrade_type=$(ynh_check_app_version_changed)
 ynh_script_progression --message="Ensuring downward compatibility..." --weight=1
 
 # If secret_key doesn't exist, create it
-if [[ -z "${secret_key:-}" ]]; then
+if [ -z "${secret_key:-}" ]; then
 	secret_key=$(ynh_string_random --length=50)
 	ynh_app_setting_set --app=$app --key=secret_key --value=$secret_key
 fi
 
-#=================================================
-# BACKUP SQLITE DATABASE AND DATAS
-#=================================================
-ynh_script_progression --message="Backup of the DB in temporary file...." --weight=5
-
-# copy data directory backup
-tmp_data_backup=$(mktemp -d)
-rsync -a $install_dir/data/ $tmp_data_backup/
-
-if [[ -f "$install_dir/config.local.user.php" ]]; then
-	cp -a $install_dir/config.local.user.php $tmp_data_backup/
-fi
-
-if  [[ $(grep "Garradin" "$tmp_data_backup/config.local.user.php") ]] ; then
-	ynh_replace_string --match_string="namespace Garradin;" --replace_string="namespace Paheko;" --target_file="$tmp_data_backup/config.local.user.php"
-fi
-
 #=================================================
 # DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
 
 if [ "$upgrade_type" == "UPGRADE_APP" ]
 then
 	ynh_script_progression --message="Upgrading source files..." --weight=5
+
+	if [ ! -d "$data_dir/data" ]; then
+    	ynh_print_warn "Migrating data directory to $data_dir..."
+
+    	# remove old configuration files
+		ynh_secure_remove --file="$install_dir/data/config.local.user.php"
+		ynh_secure_remove --file="$install_dir/config.local.yunohost.php"
+		
+		mv "$install_dir/data" "$data_dir"
+	fi
+
 	# Download, check integrity, uncompress and patch the source from app.src
-	ynh_setup_source --dest_dir="$install_dir" --full_replace=1
+	ynh_setup_source --dest_dir="$install_dir" #--keep="config.local.php"
```

Comment:
```suggestion
	ynh_setup_source --dest_dir="$install_dir" --keep="config.local.php"
```

I don't understand why there is so much confusion ... app conf files should not be savagely overwritten during `ynh_setup_source` during upgrade, we should just keep it x_x ...

Reply:
Because on last versions this file is not good ! Since version 1.2.8~ynh1
https://github.com/YunoHost-Apps/paheko_ynh/blob/29d31e91b64f9611129ba7622797113d60502f2e/conf/config.local.php#L1-L9

Reply:
If you let's `--keep="config.local.php"` it will break with preview versions...

Reply:
Sorry for the mistake, it works like this...